src/vs/workbench/contrib/notebook/browser/notebookEditorWidget.ts

```diff
@@ -2447,8 +2447,12 @@ export class NotebookEditorWidget extends Disposable implements INotebookEditorD
 				matchMap[match.cell.id] = match;
 			}
 
-			if (match.cell.cellKind === CellKind.Markup && options.includeMarkupInput) {
-				matchMap[match.cell.id] = match;
+			if (match.cell.cellKind === CellKind.Markup) {
+				if (options.markupHybrid && match.cell.getEditState() === CellEditState.Editing) {
+					matchMap[match.cell.id] = match;
+				} else if (options.includeMarkupInput) {
+					matchMap[match.cell.id] = match;
+				}
```

Comment:
nit: This repeats the same assignment several times which increases the risk of them getting out of sync. I'd do this by either combining all the ifs or track whether to set in a boolean and then setting it at the end.